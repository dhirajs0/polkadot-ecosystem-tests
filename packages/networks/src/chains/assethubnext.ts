import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNext: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNext) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNext = defineChain({
  name: 'assetHubNext',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNext,
  initStorages: getInitStorages(custom.assetHubNext),
})
