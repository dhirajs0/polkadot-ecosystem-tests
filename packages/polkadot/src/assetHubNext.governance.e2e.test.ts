import { assetHubNext } from '@e2e-test/networks/chains'

import { governanceE2ETests } from '@e2e-test/shared'

governanceE2ETests(assetHubNext, { testSuiteName: 'AHN Governance', addressEncoding: 0 })
