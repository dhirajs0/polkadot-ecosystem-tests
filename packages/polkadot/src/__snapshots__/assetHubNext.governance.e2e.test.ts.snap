// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AHN Governance > preimage submission, query and removal works > note preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`AHN Governance > preimage submission, query and removal works > unnote preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Cleared",
    "section": "preimage",
  },
]
`;

exports[`AHN Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum info before decision deposit 1`] = `
{
  "ongoing": {
    "deciding": null,
    "decisionDeposit": null,
    "enactment": {
      "after": 1,
    },
    "inQueue": false,
    "origin": {
      "origins": "SmallTipper",
    },
    "proposal": {
      "inline": "0x60030700e40b5402008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48",
    },
    "submissionDeposit": {
      "amount": 30000000000,
      "who": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "tally": {
      "ayes": 0,
      "nays": 0,
      "support": 0,
    },
    "track": 30,
  },
}
`;

exports[`AHN Governance > referendum lifecycle test - submission, decision deposit, various voting should all work > referendum submission events 1`] = `
[
  {
    "data": {
      "proposal": {
        "Inline": "(hash)",
      },
      "track": 30,
    },
    "method": "Submitted",
    "section": "referenda",
  },
]
`;
